File: go/bin_2_hex_test.go
package pehape

import (
	"testing"
)

func TestBin2Hex(t *testing.T) {
	type test struct {
		param  string
		expect string
	}
	t.Run("It should encode string into correct value", func(t *testing.T) {
		tt := &test{
			param:  "Hello World!!",
			expect: "48656c6c6f20576f726c642121",
		}

		if res := Bin2Hex(tt.param); res != tt.expect {
			t.Errorf("expected %s, but got %s", tt.expect, res)
		}
	})

	t.Run("It should encode string into correct value", func(t *testing.T) {
		tt := &test{
			param:  "hacktoberfest1234!@#",
			expect: "6861636b746f6265726665737431323334214023",
		}

		if res := Bin2Hex(tt.param); res != tt.expect {
			t.Errorf("expected %s, but got %s", tt.expect, res)
		}
	})
}

File: rust/tests/string_test/bin_2_hex_test.rs
#[cfg(test)]
mod tests {
    use pehape_rs::bin_2_hex;

    #[test]
    fn test_bin_2_hex() {
        assert_eq!(bin_2_hex("Hello World!!"), Ok(String::from("48656c6c6f20576f726c642121")));
        assert_eq!(bin_2_hex("hacktoberfest1234!@#"), Ok(String::from("6861636b746f6265726665737431323334214023")));
    }
}